momento-sdk/src/main/java/PreviewStorageClientDriver.java

```diff
@@ -0,0 +1,89 @@
+import momento.sdk.IPreviewStorageClient;
+import momento.sdk.PreviewStorageClient;
+import momento.sdk.exceptions.ClientSdkException;
+import momento.sdk.responses.storage.control.CreateStoreResponse;
+import momento.sdk.responses.storage.control.ListStoresResponse;
+import momento.sdk.responses.storage.data.GetResponse;
+
+public class PreviewStorageClientDriver {
+  public static void main(String[] args) {
+    // Instantiating the client will be largely the same as compared to cache/topics.
+    IPreviewStorageClient client = new PreviewStorageClient();
+
+    // Create a store
+    CreateStoreResponse createResponse = client.createStore("myStore").join();
+    if (createResponse instanceof CreateStoreResponse.Success) {
+      System.out.println("Store created successfully");
+    } else {
+      // Would inspect for Already exists, etc.
+      System.out.println("Error creating store");
+    }
+
+    // List all stores
+    final ListStoresResponse listResponse = client.listStores().join();
+    if (listResponse instanceof ListStoresResponse.Success) {
+      ListStoresResponse.Success success = (ListStoresResponse.Success) listResponse;
+      System.out.println("Stores:");
+      success.getStores().forEach(store -> System.out.println("- Store: " + store.getName()));
+    } else if (listResponse instanceof ListStoresResponse.Error) {
+      System.out.println(
+          "Error listing stores: " + ((ListStoresResponse.Error) listResponse).getMessage());
+    } else {
+      System.out.println("Unknown response type");
+    }
+
+    // Set a value in the store
+    client.set("myStore", "myKey", "myValue").join();
+
+    // Get the value from the store
+    final GetResponse stringResponse = client.get("myStore", "string").join();
+    printGenericGetResponse(stringResponse);
+
+    final GetResponse doubleResponse = client.get("myStore", "double").join();
+    printGenericGetResponse(doubleResponse);
+
+    if (stringResponse instanceof GetResponse.Success) {
```

Comment:
This might be a @nand4011 question but I'm trying to remember if we found a nicer way to do these due to pattern matching improvements in later versions of Java, and whether we have a way to make that visible to users?

Reply:
There is, but since this file is in the `momento.sdk` part of the repo, it's Java 8. The syntax is identical to the one in .NET: https://github.com/momentohq/client-sdk-java/blob/main/examples/cache/src/main/java/momento/client/example/BasicExample.java#L68-L76

Reply:
okay so the idea is that this file can move into the examples dir once there is a release, and we will have the option of showing the more modern patterns?

Reply:
The patterns we use in the examples are were introduced in Java 14.

Reply:
@nand4011 btw I was meaning to ask you where is the spot where you configured the examples to use 14? I couldn't find it. I was considering experimenting with making some of them use 21.